src/main/java/io/github/railroad/project/Project.java

```diff
@@ -2,19 +2,29 @@
 
 import javafx.embed.swing.SwingFXUtils;
 import javafx.scene.image.Image;
+import java.util.UUID;
+
 
 import java.awt.*;
 import java.awt.image.BufferedImage;
 import java.nio.file.Path;
-import java.util.List;
 import java.util.*;
 import java.util.concurrent.ThreadLocalRandom;
 
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+
+
 public class Project {
     private final Path path;
     private String alias;
     private Optional<Image> icon;
-    private long lastOpened = ThreadLocalRandom.current().nextLong(0, System.currentTimeMillis()); // TODO: Implement last opened
+    private long lastOpened;
+
+    private String ID;
```

Comment:
```suggestion
    private String id;
```
Variable names should be lowercase